include/drivers/clock_control.h

```diff
@@ -48,6 +47,23 @@ typedef void (*clock_control_cb_t)(struct device *dev, void *user_data);
  * INTERNAL_HIDDEN @endcond
  */
 
+/**
+ * clock_control_subsys_t is a type to identify a clock controller sub-system.
+ * Such data pointed is opaque and relevant only to the clock controller
+ * driver instance being used.
+ */
+typedef void *clock_control_subsys_t;
+
+/** @brief Callback called on clock started.
+ *
+ * @param dev		Device structure whose driver controls the clock.
+ * @param subsys	Opaque data representing the clock.
```

Comment:
"...representing the clock subsystem".  Otherwise we're getting confused between the clock and the specific subsystem that's being started.